quickwit/quickwit-serve/src/lib.rs

```diff
@@ -276,6 +305,52 @@ pub async fn serve_quickwit(config: QuickwitConfig) -> anyhow::Result<()> {
     Ok(())
 }
 
+#[derive(Clone)]
+struct RateModulator<R> {
+    rate: R,
+    memory_capacity: MemoryCapacity,
+}
+
+impl<R> RateModulator<R> {
+    pub fn new(rate: R, memory_capacity: MemoryCapacity) -> Self {
+        Self {
+            rate,
+            memory_capacity,
+        }
+    }
+}
+
+impl<R> Rate for RateModulator<R>
+where R: Rate
+{
+    fn work(&self) -> u64 {
+        let memory_usage_ratio = self.memory_capacity.usage_ratio();
+        let work = self.rate.work();
```

Comment:
The tricky part is that ideally we would like work to be a "max indexing throughpput estimator", however right now it is a actual_throughput estimator.

If we don't have any traffic, it drops to 0.

The obvious quickfix is to 

```
let work = self.rate.work().max(MIN_WORK);
````

And if you have nicer ways to measure the maximum throughput it would be better.